src/kakarot/accounts/contract/contract_account.cairo

```diff
@@ -88,3 +88,21 @@ func is_initialized{
 }() -> (is_initialized: felt) {
     return ContractAccount.is_initialized();
 }
+
+// @notice This function is used to read the nonce from storage
+// @return nonce: The current nonce of the contract account
+@view
+func get_nonce{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (nonce: felt) {
+    return ContractAccount.get_nonce();
+}
+
+// @notice This function increases the contract accounts nonce by 1
+// @return nonce: The new nonce of the contract account
+@external
+func increment_nonce{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+    nonce: felt
+) {
+    Ownable.assert_only_owner();
+    ContractAccount.increment_nonce();
```

Comment:
your `increment_nonce` implementation already return the incremented nonce, so why calling `get_nonce` again?

Reply:
Good catch, let me remove this